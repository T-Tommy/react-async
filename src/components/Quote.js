import React from 'react';
import PropTypes from 'prop-types';

function Quote({ character, quote, image }) {
  return (
    <div>
      <p>{ character }</p>
      <img src={ image } />
      <p>{ quote }</p>
    </div>
  );
}

Quote.propTypes = {
  quote: PropTypes.string,
  character: PropTypes.string,
  image: PropTypes.string
};

export default Quote;
