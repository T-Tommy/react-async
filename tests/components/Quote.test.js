import React from 'react';
import { shallow } from 'enzyme';
import Quote from '../../src/components/Quote';

describe('Quote component', () => {
  it('renders the Quote', () => {
    const wrapper = shallow(<Quote character="Fry" quote="quote" image="imgurl" />);
    expect(wrapper).toMatchSnapshot();
  });
});
