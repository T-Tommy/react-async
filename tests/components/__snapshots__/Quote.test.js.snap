// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote component renders the Quote 1`] = `
<div>
  <p>
    Fry
  </p>
  <img
    src="imgurl"
  />
  <p>
    quote
  </p>
</div>
`;
